adapters/gumgum/gumgum.go

```diff
@@ -145,6 +145,20 @@ func preprocess(imp *openrtb2.Imp) (*openrtb_ext.ExtImpGumGum, error) {
 		return nil, err
 	}
 
+    // Extract adunitid from imp.Ext
+	var adUnitID string
+	var extMap map[string]interface{}
+	if err := json.Unmarshal(imp.Ext, &extMap); err == nil {
+		if value, ok := extMap["adunitid"].(string); ok {
+			adUnitID = value
+		}
+	}
+
+	// Set adunitid to imp.TagID
+	if adUnitID != "" {
+		imp.TagID = adUnitID
+	}
+
```

Comment:
issue: please provide test coverage for this new functionality via the json testing framework